### PR TITLE
Require Kokkos 4.0

### PR DESCRIPTION
**Pull request by aprokop**
- Require Kokkos 4.0
- Switch 3.7 builds to 4.1
- Remove pre Kokkos 4.0 workarounds

### CODE REVIEW

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
retest this pleaae

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
 project(ArborX CXX)
 
-# Workaround to compile against Trilinos 14.0. That version
-# of Trilinos does not set the compatibility mode for Kokkos correctly.
-# FIXME: remove once we require Kokkos 4.0.1 or later.
-find_package(Kokkos QUIET 3.7.01 CONFIG)
-if (NOT Kokkos_FOUND)
-  find_package(Kokkos 4.0.00 REQUIRED CONFIG)
-endif()
+find_package(Kokkos 4.0.00 REQUIRED CONFIG)
```
Comment (dalg24) [inline CMakeLists.txt:4]:
```suggestion
find_package(Kokkos 4.0 REQUIRED CONFIG)
```

**Comment by masterleinad** [comment]
We can also replace `KOKKOS_VERSION` comparisons with `KOKKOS_VERSION_GREATER_EQUAL(MAJOR, MINOR, PATCH)` from `Kokkos` 4.0.00 on, see https://github.com/kokkos/kokkos/pull/5424.

**Comment by aprokop** [comment]
> We can also replace `KOKKOS_VERSION` comparisons with `KOKKOS_VERSION_GREATER_EQUAL(MAJOR, MINOR, PATCH)` from `Kokkos` 4.0.00 on, see [kokkos/kokkos#5424](https://github.com/kokkos/kokkos/pull/5424).

Imho, this seems to be more complicated than `KOKKOS_VERSION >= 40100`.

**Comment by masterleinad** [comment]
> Imho, this seems to be more complicated than `KOKKOS_VERSION >= 40100`.

The main motivation is that the new syntax clearly separates major, minor, and patch versions whereas that is not so clear for `40100`.

**Comment by aprokop** [comment]
> The main motivation is that the new syntax clearly separates major, minor, and patch versions whereas that is not so clear for `40100`.

I understand that. It's a tradeoff.

**Pull request merged**